src/Engine/EngineInterface.php

```diff
@@ -10,6 +10,15 @@
 
 interface EngineInterface
 {
+
+    /**
+     * Requires the Analytics instance which handles the Events
+     * that occur during the test process.
+     * 
+     * @param AnalyticsInterface $analytics
+     */
+    public function __construct(AnalyticsInterface $analytics);
```

Comment:
I guess that was the entire idea behind reinforcing the contract with the implementation.
To be honest, I thought that's what we agreed on :)